RecentActivity/src/org/sleuthkit/autopsy/recentactivity/ExtractSafari.java

```diff
@@ -636,7 +641,19 @@ private BlackboardArtifact parseDownloadDictionary(Content dataSource, AbstractF
 
         BlackboardArtifact bbart = origFile.newArtifact(BlackboardArtifact.ARTIFACT_TYPE.TSK_WEB_DOWNLOAD);
         bbart.addAttributes(this.createDownloadAttributes(path, pathID, url, time, NetworkUtils.extractDomain(url), getName()));
-
-        return bbart;
+        bbartifacts.add(bbart);
+        
+        // find the downloaded file and create a TSK_DOWNLOAD_SOURCE for it.
+        for (AbstractFile downloadedFile : fileManager.findFiles(dataSource, FilenameUtils.getName(path), FilenameUtils.getPath(path))) {
```

Comment:
Is this the same for all the browsers?  Could you please add this to the createDownloadAttributes instead of this function?   Someday I would like to roll the createXXXAttributes out to all the extract subclasses.